src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> # prepare image and text
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "hello world"
+        >>> processor = BridgeTowerProcessor.from_pretrained("BridgeTower/bridgetower-base")
+        >>> model = BridgeTowerModel.from_pretrained("BridgeTower/bridgetower-base")
+
+        >>> inputs = processor(image, text, return_tensors="pt")
+        >>> outputs = model(**inputs)
+        >>> outputs.keys()
+        odict_keys(['text_feats', 'image_feats', 'pooler_output'])
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        image_token_type_idx = image_token_type_idx if image_token_type_idx else 1
+        irtr_len = 0
+        input_shape = input_ids.size()
+        text_embeds = self.text_transformer.embeddings(input_ids=input_ids)
+
+        if attention_mask is None:
+            attention_mask = torch.ones(input_shape, dtype=torch.long, device=self.device)
+        extend_text_masks = self.text_transformer.get_extended_attention_mask(attention_mask, input_shape, self.device)
+
+        split_index = len(self.text_transformer.encoder.layer) - self.config.num_hidden_layers + 1
+        for layer in self.text_transformer.encoder.layer[:split_index]:
+            text_embeds = layer(text_embeds, extend_text_masks)[0]
+
+        image_embeds = self.vit_model.visual.forward_pre(pixel_values.type(self.vit_model.dtype))
+        for block in self.vit_model.visual.transformer.resblocks[:split_index]:
+            image_embeds = block(image_embeds)
+        image_embeds_with_ln = self.vit_model.visual.forward_post(image_embeds.type(self.vit_model.dtype))
+
+        # first layer
+        cross_modal_text = self.cross_modal_text_transform(text_embeds)
+        text_token_type_embeddings = self.token_type_embeddings(torch.zeros(1).long().to(self.device)).expand_as(
+            cross_modal_text
+        )
+        cross_modal_text = self.cross_modal_text_layernorm(cross_modal_text + text_token_type_embeddings)
+
+        image_embeds_with_ln = self.cross_modal_image_transform(image_embeds_with_ln)
+        image_token_type_embeddings = self.token_type_embeddings(
+            torch.zeros(1).long().to(self.device).fill_(image_token_type_idx)
+        ).expand_as(image_embeds_with_ln)
+        image_embeds_with_ln = image_embeds_with_ln + image_token_type_embeddings
+        cross_modal_image = self.cross_modal_image_layernorm(image_embeds_with_ln)
+        if irtr_len > 0:
+            _bs, _L, _D = image_embeds_with_ln.size()
+            cross_modal_image = cross_modal_image.unsqueeze(1).expand(_bs, irtr_len, _L, _D).reshape(-1, _L, _D)
+        pixel_mask = torch.ones(
+            (cross_modal_image.size(0), cross_modal_image.size(1)), dtype=torch.long, device=self.device
+        )
```

Comment:
```suggestion
        pixel_mask = torch.ones(_bs * irtrt_len, _L), dtype=torch.long, device=self.device)
```

Reply:
The suggested changes does not provide the same results and gave me a few shape mismatch errors. I've retained the original code.

Reply:
Again here.